Add breadcrumbs documentation (#3202)
SAP/luigi: docs/navigation-advanced.md

  ...
```

## Breadcrumbs

> Luigi allows you to add [breadcrumbs](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Breadcrumb_Navigation) to your application. First, you need to create your own custom code implementing breadcrumbs, then enable them on the node level using Luigi's [showBreadcrumbs](navigation-parameters-reference.md#showbreadcrumbs) parameter. In your custom code, you can choose any look and style for the breadcrumbs as well as define what should happen upon clicking them. However, the code should follow this general pattern and return the variable `breadcrumbs`: 

Reviewer: ```suggestion
Luigi allows you to add [breadcrumbs](https://developer.mozilla.org/en-US/docs/Web/CSS/Layout_cookbook/Breadcrumb_Navigation) to your application. You need to create your own custom code implementing breadcrumbs. Once the breadcrumbs config is set it is enabled by default for all nodes. If you wish to disable it for a particular node you need to set [showBreadcrumbs](navigation-parameters-reference.md#showbreadcrumbs): true for that node.
The breadcrumbs configuration structure looks like this:

      navigation.breadcrumbs = {
            omitRoot: false,  // if set to true, the root node in breadcrumb hierarchy is omitted 
            clearBeforeRender: true, // if set to true, the containerElement will be cleared first, before being rendered. If set to false, handling of the clear before render needs to be handled by your side 
            autoHide: true, // hide breadcrumbs when navigating to root node
            renderer: (containerElement, nodeItems, clickHandler)  => {
                  // containerElement - refers to HTML element that contains the breadcrumb structure to which you can append your own customised elements
                  // nodeItems : {
                       label: label of node,
                       node: navigation node,
                       route: node route,
                       pending: indicates whether node label has not been resolved yet or not
                  }
                  // clickHandler(node) , can be called per node, to navigate to that node
            }
      }

. In your custom code, you can choose any look and style for the breadcrumbs as well as define what should happen upon clicking them. However, the code should follow this general pattern and return the variable `breadcrumbs`: 
```

```js
var breadcrumbsConfig = {
      clearBeforeRender: true,
      renderer: (el, items, clickHandler) => {
       // custom code defining how breadcrumbs should look like
       // let breadcrumbs = ...
       return breadcrumbs
      }
}
```

<!-- accordion:start -->
Below is an example of a simple `breadcrumbsConfig`:

### Click to expand

```js

> var breadcrumbsConfig = {

Reviewer: ```suggestion
navigation.breadcrumbs = {
```

            itemCmp.setAttribute('style', 'display:inline; margin: 0 10px;');
            itemCmp.setAttribute('data-testid', `breadcrumb_${item.label}_index${index}`);
            itemCmp.innerHTML = item.label;
            itemCmp._item = item;
            breadcrumbs.appendChild(itemCmp);
          }
        });
        breadcrumbs.addEventListener('click', event => {
          console.log('event detail', event);
          event.preventDefault();
          clickHandler(event.detail.item._item);
        });
        el.appendChild(breadcrumbs);
        return breadcrumbs;
      }
    };
```

Below is another example which uses UI5 breadcrumbs: 

### Click to expand

```js
config.navigation.breadcrumbs = {
            autoHide: true,
            omitRoot: false,
            pendingItemLabel: '...',
            renderer: (el, items, clickHandler) => {
              el.classList.add('dxp-breadcrumb');
              const ui5breadcrumbs =
                el.querySelector('ui5-breadcrumbs') ||
                document.createElement('ui5-breadcrumbs');
              ui5breadcrumbs.innerHTML = '';
              items.forEach((item, index) => {

>                 const label: string = item.label;

Reviewer: ```suggestion
                const label = item.label;
```

                
                if (label && !label.startsWith(':virtualSegment_')) {
                  const itemCmp = document.createElement(

>                     'ui5-breadcrumbs-item',

Reviewer: ```suggestion
                    'ui5-breadcrumbs-item'
```

                  );
                  itemCmp.setAttribute('href', item.route);
                  itemCmp.innerHTML = label;

>                   (itemCmp as any)._item = item;

Reviewer: ```suggestion
                  itemCmp._item = item;
```

                  ui5breadcrumbs.appendChild(itemCmp);
                }
              });

>               ui5breadcrumbs.addEventListener('item-click', (event: any) => {

Reviewer: ```suggestion
              ui5breadcrumbs.addEventListener('item-click', (event) => {
```

            }
      }
```

<!-- accordion:start -->
Below is an example of a simple `breadcrumbsConfig`:

### Click to expand

```js
navigation.breadcrumbs = {
      clearBeforeRender: true,
      renderer: (el, items, clickHandler) => {
        el.classList.add('myBreadcrumb');
        let breadcrumbs = document.createElement('ol');
        breadcrumbs.setAttribute('style', 'top: 0;position: absolute;left: 0;');
        items.forEach((item, index) => {
          if (item.label) {
            let itemCmp = document.createElement('li');
            itemCmp.setAttribute('style', 'display:inline; margin: 0 10px;');
            itemCmp.setAttribute('data-testid', `breadcrumb_${item.label}_index${index}`);
            itemCmp.innerHTML = item.label;
            itemCmp._item = item;
            breadcrumbs.appendChild(itemCmp);
          }
        });
        breadcrumbs.addEventListener('click', event => {
          console.log('event detail', event);
          event.preventDefault();
          clickHandler(event.detail.item._item);
        });
        el.appendChild(breadcrumbs);
        return breadcrumbs;
      }
    };
```

> Below is another example which uses UI5 breadcrumbs: 

Reviewer 2: ... uses UI5 Web Components breadcrumbs:

> Below is another example which uses UI5 breadcrumbs: 

Author: ```suggestion
Below is another example which uses UI5 Web Components breadcrumbs: 
```
